### PR TITLE
Companion for Use proper bounded vector type for nominations #10601

### PR DESCRIPTION
**Pull request by georgesdib**
Companion to paritytech/substrate#10601

Renaming `MAX_NOMINATIONS` to `MaxNominations` and changing type after changes in substrate in the accompanying PR.

@kianenigma

### CODE REVIEW

File: runtime/kusama/src/lib.rs [hunk 1]
```diff
@@ -532,6 +532,7 @@ parameter_types! {
 	pub const SlashDeferDuration: pallet_staking::EraIndex = 27;
 	pub const MaxNominatorRewardedPerValidator: u32 = 256;
 	pub const OffendingValidatorsThreshold: Perbill = Perbill::from_percent(17);
+	pub const MaxNominations: u32 = <NposCompactSolution24 as sp_npos_elections::NposSolution>::LIMIT as u32;
```
Comment (kianenigma) [inline runtime/kusama/src/lib.rs:535]:
```suggestion
	// 24
	pub const MaxNominations: u32 = <NposCompactSolution24 as sp_npos_elections::NposSolution>::LIMIT as u32;
```

File: runtime/polkadot/src/lib.rs [hunk 2]
```diff
@@ -528,6 +528,7 @@ parameter_types! {
 	pub const RewardCurve: &'static PiecewiseLinear<'static> = &REWARD_CURVE;
 	pub const MaxNominatorRewardedPerValidator: u32 = 256;
 	pub const OffendingValidatorsThreshold: Perbill = Perbill::from_percent(17);
+	pub const MaxNominations: u32 = <NposCompactSolution16 as sp_npos_elections::NposSolution>::LIMIT as u32;
```
Comment (kianenigma) [inline runtime/polkadot/src/lib.rs:531]:
```suggestion
	// 16
	pub const MaxNominations: u32 = <NposCompactSolution16 as sp_npos_elections::NposSolution>::LIMIT as u32;
```

**Review by kianenigma — APPROVED** [review]

**Comment by kianenigma** [comment]
hmm strange companion build failure here.

**Comment by georgesdib** [comment]
> hmm strange companion build failure here. 

Was the build retriggered after my merge for the substrate PR?

**Comment by georgesdib** [comment]
The status is a bit weird, what does that mean?

**Comment by bkchr** [comment]
> The status is a bit weird, what does that mean? Is it stuck checking if the substrate PR merged well? (That PR seems to be in a weird state as well, having successfully merged but then returning a merge error)

It required to approve running the jobs, because this was the first time you created a pr.

**Pull request merged**